In [ ]:
import numpy as np
import pandas as pd
import seaborn as sb
import matplotlib.pyplot as plt
sb.set()

from sklearn.impute import KNNImputer

# **Big Dataset 1**



> *Dataframe: BigData1*





In [ ]:
#Importing Big Dataset 1
BigData1 = pd.read_csv('Big Dataset_1.csv')
BigData1.head()

Diabetes_binary  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0              0.0     1.0       1.0        1.0  40.0     1.0     0.0   
1              0.0     0.0       0.0        0.0  25.0     1.0     0.0   
2              0.0     1.0       1.0        1.0  28.0     0.0     0.0   
3              0.0     1.0       0.0        1.0  27.0     0.0     0.0   
4              0.0     1.0       1.0        1.0  24.0     0.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           0.0     0.0  ...            1.0   
1                   0.0           1.0     0.0  ...            0.0   
2                   0.0           0.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      5.0      18.0      15.0       1.0  0.0   9.0        4.0   
1          1.0      3.0       0.0       0.0       0.0  0.0   7.0        6.0   
2          1.0      5.0      30.0      30.0       1.0  0.0   9.0        4.0   
3          0.0      2.0       0.0       0.0       0.0  0.0  11.0        3.0   
4          0.0      2.0       3.0       0.0       0.0  0.0  11.0        5.0   

   Income  
0     3.0  
1     1.0  
2     8.0  
3     6.0  
4     4.0  

[5 rows x 22 columns]

In [ ]:
print("BigData1 dimensions : ", BigData1.shape)
print()
BigData1.info()

BigData1 dimensions :  (253680, 22)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 253680 entries, 0 to 253679
Data columns (total 22 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_binary       253680 non-null  float64
 1   HighBP                253680 non-null  float64
 2   HighChol              253680 non-null  float64
 3   CholCheck             253680 non-null  float64
 4   BMI                   253680 non-null  float64
 5   Smoker                253680 non-null  float64
 6   Stroke                253680 non-null  float64
 7   HeartDiseaseorAttack  253680 non-null  float64
 8   PhysActivity          253680 non-null  float64
 9   Fruits                253680 non-null  float64
 10  Veggies               253680 non-null  float64
 11  HvyAlcoholConsump     253680 non-null  float64
 12  AnyHealthcare         253680 non-null  float64
 13  NoDocbcCost           253680 non-null  float64
 14  GenHlth        

In [ ]:
BigData1.nunique()

Diabetes_binary          2
HighBP                   2
HighChol                 2
CholCheck                2
BMI                     84
Smoker                   2
Stroke                   2
HeartDiseaseorAttack     2
PhysActivity             2
Fruits                   2
Veggies                  2
HvyAlcoholConsump        2
AnyHealthcare            2
NoDocbcCost              2
GenHlth                  5
MentHlth                31
PhysHlth                31
DiffWalk                 2
Sex                      2
Age                     13
Education                6
Income                   8
dtype: int64

# **Big Dataset 2**


> *Dataframe: BigData2*



In [ ]:
BigData2 = pd.read_csv('Big Dataset_2.csv')
BigData2.head()

Diabetes_binary  HighBP  HighChol  CholCheck   BMI  Smoker  Stroke  \
0              0.0     1.0       0.0        1.0  26.0     0.0     0.0   
1              0.0     1.0       1.0        1.0  26.0     1.0     1.0   
2              0.0     0.0       0.0        1.0  26.0     0.0     0.0   
3              0.0     1.0       1.0        1.0  28.0     1.0     0.0   
4              0.0     0.0       0.0        1.0  29.0     1.0     0.0   

   HeartDiseaseorAttack  PhysActivity  Fruits  ...  AnyHealthcare  \
0                   0.0           1.0     0.0  ...            1.0   
1                   0.0           0.0     1.0  ...            1.0   
2                   0.0           1.0     1.0  ...            1.0   
3                   0.0           1.0     1.0  ...            1.0   
4                   0.0           1.0     1.0  ...            1.0   

   NoDocbcCost  GenHlth  MentHlth  PhysHlth  DiffWalk  Sex   Age  Education  \
0          0.0      3.0       5.0      30.0       0.0  1.0   4.0        6.0   
1          0.0      3.0       0.0       0.0       0.0  1.0  12.0        6.0   
2          0.0      1.0       0.0      10.0       0.0  1.0  13.0        6.0   
3          0.0      3.0       0.0       3.0       0.0  1.0  11.0        6.0   
4          0.0      2.0       0.0       0.0       0.0  0.0   8.0        5.0   

   Income  
0     8.0  
1     8.0  
2     8.0  
3     8.0  
4     8.0  

[5 rows x 22 columns]

In [ ]:
print("BigData2 dimensions : ", BigData2.shape)
print()
BigData2.info()

BigData2 dimensions :  (70692, 22)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70692 entries, 0 to 70691
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Diabetes_binary       70692 non-null  float64
 1   HighBP                70692 non-null  float64
 2   HighChol              70692 non-null  float64
 3   CholCheck             70692 non-null  float64
 4   BMI                   70692 non-null  float64
 5   Smoker                70692 non-null  float64
 6   Stroke                70692 non-null  float64
 7   HeartDiseaseorAttack  70692 non-null  float64
 8   PhysActivity          70692 non-null  float64
 9   Fruits                70692 non-null  float64
 10  Veggies               70692 non-null  float64
 11  HvyAlcoholConsump     70692 non-null  float64
 12  AnyHealthcare         70692 non-null  float64
 13  NoDocbcCost           70692 non-null  float64
 14  GenHlth               70692 non-nu

In [ ]:
BigData2.nunique()

Diabetes_binary          2
HighBP                   2
HighChol                 2
CholCheck                2
BMI                     80
Smoker                   2
Stroke                   2
HeartDiseaseorAttack     2
PhysActivity             2
Fruits                   2
Veggies                  2
HvyAlcoholConsump        2
AnyHealthcare            2
NoDocbcCost              2
GenHlth                  5
MentHlth                31
PhysHlth                31
DiffWalk                 2
Sex                      2
Age                     13
Education                6
Income                   8
dtype: int64

## **Concating Big Datasets**


> *Dataframe: BigDataConcat*




> Concating BigData1 (253680 datapoints) and BigData2 (70692 datapoints) which have the same columns (variables) to get a larger sample of datapoints.



In [ ]:
BigDataConcat = pd.concat([BigData1, BigData2])

## **Cleaning BigDataConcat**

In [ ]:
#Renaming Age as AgeLevel
BigDataConcat.rename(columns={'Age':'AgeLevel'}, inplace=True)

In [ ]:
#Removing unnecessary variables
print("BigDataConcat Dimensions before dropping variables: ", BigDataConcat.shape) 

to_drop = ['HighBP', 'CholCheck', 'Fruits', 'Veggies', 'AnyHealthcare', 'NoDocbcCost', 'GenHlth', 'Sex', 'Education', 'Income'] 
BigDataConcat.drop(to_drop, inplace=True, axis=1)

print("BigDataConcat Dimensions after dropping variables: ", BigDataConcat.shape)

BigDataConcat Dimensions before dropping variables:  (324372, 22)
BigDataConcat Dimensions after dropping variables:  (324372, 12)


In [ ]:
#Removing duplicate rows
print("BigDataConcat Dimensions before dropping duplicate rows: ", BigDataConcat.shape) 
print()
print("Number of duplicate rows: ", BigDataConcat.duplicated().sum())
BigDataConcat = BigDataConcat.drop_duplicates()
print()
print("BigDataConcat Dimensions after dropping duplicate rows: ", BigDataConcat.shape)

BigDataConcat Dimensions before dropping duplicate rows:  (324372, 12)

Number of duplicate rows:  218651

BigDataConcat Dimensions after dropping duplicate rows:  (105721, 12)


In [ ]:
BigDataConcat.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 105721 entries, 0 to 253676
Data columns (total 12 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   Diabetes_binary       105721 non-null  float64
 1   HighChol              105721 non-null  float64
 2   BMI                   105721 non-null  float64
 3   Smoker                105721 non-null  float64
 4   Stroke                105721 non-null  float64
 5   HeartDiseaseorAttack  105721 non-null  float64
 6   PhysActivity          105721 non-null  float64
 7   HvyAlcoholConsump     105721 non-null  float64
 8   MentHlth              105721 non-null  float64
 9   PhysHlth              105721 non-null  float64
 10  DiffWalk              105721 non-null  float64
 11  AgeLevel              105721 non-null  float64
dtypes: float64(12)
memory usage: 10.5 MB


# **Small Dataset**


> *Dataframe: SmallData*



In [ ]:
SmallData = pd.read_csv('Small Dataset.csv')
SmallData.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            2      138             62             35        0  33.6   
1            0       84             82             31      125  38.2   
2            0      145              0              0        0  44.2   
3            0      135             68             42      250  42.3   
4            1      139             62             41      480  40.7   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.127   47        1  
1                     0.233   23        0  
2                     0.630   31        1  
3                     0.365   24        1  
4                     0.536   21        0

In [ ]:
print("SmallData dimensions : ", SmallData.shape)
print()
SmallData.info()

SmallData dimensions :  (2000, 9)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Pregnancies               2000 non-null   int64  
 1   Glucose                   2000 non-null   int64  
 2   BloodPressure             2000 non-null   int64  
 3   SkinThickness             2000 non-null   int64  
 4   Insulin                   2000 non-null   int64  
 5   BMI                       2000 non-null   float64
 6   DiabetesPedigreeFunction  2000 non-null   float64
 7   Age                       2000 non-null   int64  
 8   Outcome                   2000 non-null   int64  
dtypes: float64(2), int64(7)
memory usage: 140.8 KB


In [ ]:
SmallData.nunique()

Pregnancies                  17
Glucose                     136
BloodPressure                47
SkinThickness                53
Insulin                     182
BMI                         247
DiabetesPedigreeFunction    505
Age                          52
Outcome                       2
dtype: int64

## **Cleaning SmallData**

In [ ]:
#add a new column ('AgeLevel') next to the the column 'Age' and group ages into age levels
Age_class = pd.cut(SmallData.Age, bins=[18,25,30,35,40,45,50,55,60,65,70,75,80,99], labels=['1','2','3','4','5','6','7','8','9','10','11','12','13'])
SmallData.insert(8,'AgeLevel', Age_class)

#drop the column Age
SmallData.drop('Age', inplace=True, axis=1)

#Renaming outcome as Diabetes_binary
SmallData.rename(columns={'Outcome':'Diabetes_binary'}, inplace=True)

In [ ]:
#Removing duplicate rows
print("SmallData Dimensions before dropping duplicate rows: ", SmallData.shape) 
print()
print("Number of duplicate rows: ", SmallData.duplicated().sum())
SmallData = SmallData.drop_duplicates()
print()
print("SmallData Dimensions after dropping duplicate rows: ", SmallData.shape)

SmallData Dimensions before dropping duplicate rows:  (2000, 9)

Number of duplicate rows:  1256

SmallData Dimensions after dropping duplicate rows:  (744, 9)


In [ ]:
#Finding the number of 0s in each of the numerical variables of SmallData (before removal)
print("Before cleaning zeros:")
print()

print("No of 0s in Glucose:", SmallData.Glucose[SmallData.Glucose==0].count())
print("No of 0s in BloodPressure:", SmallData.BloodPressure[SmallData.BloodPressure==0].count())
print("No of 0s in SkinThickness:",SmallData.SkinThickness[SmallData.SkinThickness==0].count())
print("No of 0s in Insulin:", SmallData.Insulin[SmallData.Insulin==0].count())
print("No of 0s in DiabetesPedigreeFunction:", SmallData.DiabetesPedigreeFunction[SmallData.DiabetesPedigreeFunction==0].count())

Before cleaning zeros:

No of 0s in Glucose: 5
No of 0s in BloodPressure: 34
No of 0s in SkinThickness: 215
No of 0s in Insulin: 359
No of 0s in DiabetesPedigreeFunction: 0


In [ ]:
#Removing all the 0s in each of the numerical variables of SmallData
SmallData =SmallData[SmallData['Glucose'] !=0]
SmallData =SmallData[SmallData['BloodPressure'] !=0]
SmallData =SmallData[SmallData['SkinThickness'] !=0]
SmallData =SmallData[SmallData['Insulin'] !=0]
SmallData =SmallData[SmallData['DiabetesPedigreeFunction'] !=0]

In [ ]:
#Finding the number of 0s in numerical variables of SmallData (after removal)
print("After cleaning zeros:")
print()

print("No of 0s in Glucose:", SmallData.Glucose[SmallData.Glucose==0].count())
print("No of 0s in BloodPressure:", SmallData.BloodPressure[SmallData.BloodPressure==0].count())
print("No of 0s in SkinThickness:",SmallData.SkinThickness[SmallData.SkinThickness==0].count())
print("No of 0s in Insulin:", SmallData.Insulin[SmallData.Insulin==0].count())
print("No of 0s in DiabetesPedigreeFunction:", SmallData.DiabetesPedigreeFunction[SmallData.DiabetesPedigreeFunction==0].count())

After cleaning zeros:

No of 0s in Glucose: 0
No of 0s in BloodPressure: 0
No of 0s in SkinThickness: 0
No of 0s in Insulin: 0
No of 0s in DiabetesPedigreeFunction: 0


# **Appending BigDataConcat and SmallData -** *JointData*



> Sampling 700 random datapoints from BigDataConcat and 300 random datapoints from SmallData and appending them to form JointData. 
Common variables between BigDataConcat and SmallData are AgeLevel, BMI and Diabetes_binary.
JointData has more variables (mix of both numerical and categorical variables) that we can use for diabetes prediction.





In [ ]:
BigDataSample = BigDataConcat.sample(n=700)
SmallDataSample = SmallData.sample(n=300)

JointData = BigDataSample.append(SmallDataSample)
print("JointData dimensions: ", JointData.shape)

JointData dimensions:  (1000, 18)


<ipython-input-109-0bdf552c3caa>:4: FutureWarning: The frame.append method is deprecated and will be removed from pandas in a future version. Use pandas.concat instead.
  JointData = BigDataSample.append(SmallDataSample)


In [ ]:
#Check for any duplicate rows
print("Duplicate Rows: ", JointData.duplicated().sum())

Duplicate Rows:  0


In [ ]:
JointData = JointData.reset_index()
JointData = JointData.drop(columns='index')
JointData.head(n=1000)

Diabetes_binary  HighChol   BMI  Smoker  Stroke  HeartDiseaseorAttack  \
0                0.0       0.0  25.0     1.0     0.0                   0.0   
1                0.0       1.0  19.0     0.0     0.0                   0.0   
2                0.0       0.0  23.0     0.0     1.0                   0.0   
3                1.0       1.0  35.0     0.0     0.0                   0.0   
4                0.0       0.0  31.0     0.0     0.0                   0.0   
..               ...       ...   ...     ...     ...                   ...   
995              0.0       NaN  43.4     NaN     NaN                   NaN   
996              1.0       NaN  40.0     NaN     NaN                   NaN   
997              0.0       NaN  28.4     NaN     NaN                   NaN   
998              0.0       NaN  26.3     NaN     NaN                   NaN   
999              1.0       NaN  28.3     NaN     NaN                   NaN   

     PhysActivity  HvyAlcoholConsump  MentHlth  PhysHlth  DiffWalk AgeLevel  \
0             1.0                0.0       0.0      15.0       0.0     10.0   
1             0.0                0.0       0.0      14.0       0.0      9.0   
2             0.0                0.0       0.0      30.0       1.0     11.0   
3             0.0                0.0       0.0       0.0       1.0     12.0   
4             1.0                0.0       2.0      10.0       0.0      1.0   
..            ...                ...       ...       ...       ...      ...   
995           NaN                NaN       NaN       NaN       NaN        3   
996           NaN                NaN       NaN       NaN       NaN        1   
997           NaN                NaN       NaN       NaN       NaN        1   
998           NaN                NaN       NaN       NaN       NaN        1   
999           NaN                NaN       NaN       NaN       NaN        5   

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0            NaN      NaN            NaN            NaN      NaN   
1            NaN      NaN            NaN            NaN      NaN   
2            NaN      NaN            NaN            NaN      NaN   
3            NaN      NaN            NaN            NaN      NaN   
4            NaN      NaN            NaN            NaN      NaN   
..           ...      ...            ...            ...      ...   
995          0.0     93.0          100.0           39.0     72.0   
996          1.0    181.0           78.0           42.0    293.0   
997          2.0     88.0           58.0           26.0     16.0   
998          3.0    116.0           74.0           15.0    105.0   
999         11.0    136.0           84.0           35.0    130.0   

     DiabetesPedigreeFunction  
0                         NaN  
1                         NaN  
2                         NaN  
3                         NaN  
4                         NaN  
..                        ...  
995                     1.021  
996                     1.258  
997                     0.766  
998                     0.107  
999                     0.260  

[1000 rows x 18 columns]

# **Prediction of null values in JointData**

## **KNN Imputation**

In [ ]:
#imputing null values using KNN Imputer
JointData_knn_Imputer = JointData.copy(deep=True)
knnimputer = KNNImputer(n_neighbors=10)
JointData_knn_Imputer.iloc[:,:] = knnimputer.fit_transform(JointData_knn_Imputer)

JointData_knn_Imputer = JointData_knn_Imputer.round({'Pregnancies': 0, 'HighChol': 0, 'Smoker': 0, 'Stroke': 0, 'PhysActivity': 0, 'HvyAlcoholConsump': 0,'HeartDiseaseorAttack': 0,'DiffWalk': 0, 'MentHlth': 0, 'PhysHlth': 0, 'BloodPressure': 0, 'SkinThickness': 0, 'Insulin': 0,})

JointData_knn_Imputer.head(n=1000)

<ipython-input-112-080c0e055738>:4: DeprecationWarning: In a future version, `df.iloc[:, i] = newvals` will attempt to set the values inplace instead of always setting a new array. To retain the old behavior, use either `df[df.columns[i]] = newvals` or, if columns are non-unique, `df.isetitem(i, newvals)`
  JointData_knn_Imputer.iloc[:,:] = knnimputer.fit_transform(JointData_knn_Imputer)


Diabetes_binary  HighChol   BMI  Smoker  Stroke  HeartDiseaseorAttack  \
0                0.0       0.0  25.0     1.0     0.0                   0.0   
1                0.0       1.0  19.0     0.0     0.0                   0.0   
2                0.0       0.0  23.0     0.0     1.0                   0.0   
3                1.0       1.0  35.0     0.0     0.0                   0.0   
4                0.0       0.0  31.0     0.0     0.0                   0.0   
..               ...       ...   ...     ...     ...                   ...   
995              0.0       0.0  43.4     0.0     0.0                   0.0   
996              1.0       0.0  40.0     0.0     0.0                   0.0   
997              0.0       0.0  28.4     0.0     0.0                   0.0   
998              0.0       0.0  26.3     0.0     0.0                   0.0   
999              1.0       1.0  28.3     1.0     0.0                   0.0   

     PhysActivity  HvyAlcoholConsump  MentHlth  PhysHlth  DiffWalk  AgeLevel  \
0             1.0                0.0       0.0      15.0       0.0      10.0   
1             0.0                0.0       0.0      14.0       0.0       9.0   
2             0.0                0.0       0.0      30.0       1.0      11.0   
3             0.0                0.0       0.0       0.0       1.0      12.0   
4             1.0                0.0       2.0      10.0       0.0       1.0   
..            ...                ...       ...       ...       ...       ...   
995           1.0                0.0      10.0       5.0       0.0       3.0   
996           1.0                0.0       7.0       4.0       0.0       1.0   
997           1.0                0.0       7.0       3.0       0.0       1.0   
998           1.0                0.0       6.0       1.0       0.0       1.0   
999           1.0                0.0      12.0       8.0       0.0       5.0   

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0            9.0    143.7           71.0           27.0    210.0   
1            5.0    134.0           70.0           25.0    122.0   
2            7.0    141.6           73.0           29.0    165.0   
3            6.0    152.5           82.0           37.0    214.0   
4            2.0    111.9           69.0           25.0    166.0   
..           ...      ...            ...            ...      ...   
995          0.0     93.0          100.0           39.0     72.0   
996          1.0    181.0           78.0           42.0    293.0   
997          2.0     88.0           58.0           26.0     16.0   
998          3.0    116.0           74.0           15.0    105.0   
999         11.0    136.0           84.0           35.0    130.0   

     DiabetesPedigreeFunction  
0                      0.5975  
1                      0.5288  
2                      0.5717  
3                      0.7138  
4                      0.6118  
..                        ...  
995                    1.0210  
996                    1.2580  
997                    0.7660  
998                    0.1070  
999                    0.2600  

[1000 rows x 18 columns]

In [ ]:
#Check for any duplicate rows
print("Duplicate Rows : ", JointData_knn_Imputer.duplicated().sum())

Duplicate Rows :  0


In [ ]:
#Check for any null values
print("Null Values: ", JointData_knn_Imputer.isna().sum().sum())

Null Values:  0


In [ ]:
#Convert variables to appropriate datatypes
JointData_knn_Imputer = JointData_knn_Imputer.astype({"Diabetes_binary":'int64',  
                                                      "HighChol":'object', "Smoker":'object', "Stroke":'object', 
                                                      "HeartDiseaseorAttack": 'object',"PhysActivity":'object', 
                                                      "HvyAlcoholConsump":'object', "DiffWalk": 'object', "AgeLevel": 'object', 
                                                      "MentHlth":'int', "PhysHlth":'int', "Pregnancies":'int',
                                                      "Glucose":'int', "BloodPressure":'int', "SkinThickness":'int',
                                                      "Insulin":'int'})

In [ ]:
JointData_knn_Imputer.head()

Diabetes_binary HighChol   BMI Smoker Stroke HeartDiseaseorAttack  \
0                0      0.0  25.0    1.0    0.0                  0.0   
1                0      1.0  19.0    0.0    0.0                  0.0   
2                0      0.0  23.0    0.0    1.0                  0.0   
3                1      1.0  35.0    0.0    0.0                  0.0   
4                0      0.0  31.0    0.0    0.0                  0.0   

  PhysActivity HvyAlcoholConsump  MentHlth  PhysHlth DiffWalk AgeLevel  \
0          1.0               0.0         0        15      0.0     10.0   
1          0.0               0.0         0        14      0.0      9.0   
2          0.0               0.0         0        30      1.0     11.0   
3          0.0               0.0         0         0      1.0     12.0   
4          1.0               0.0         2        10      0.0      1.0   

   Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0            9      143             71             27      210   
1            5      134             70             25      122   
2            7      141             73             29      165   
3            6      152             82             37      214   
4            2      111             69             25      166   

   DiabetesPedigreeFunction  
0                    0.5975  
1                    0.5288  
2                    0.5717  
3                    0.7138  
4                    0.6118

In [ ]:
print(JointData_knn_Imputer.dtypes)

Diabetes_binary               int64
HighChol                     object
BMI                         float64
Smoker                       object
Stroke                       object
HeartDiseaseorAttack         object
PhysActivity                 object
HvyAlcoholConsump            object
MentHlth                      int64
PhysHlth                      int64
DiffWalk                     object
AgeLevel                     object
Pregnancies                   int64
Glucose                       int64
BloodPressure                 int64
SkinThickness                 int64
Insulin                       int64
DiabetesPedigreeFunction    float64
dtype: object


In [ ]:
#Final, cleaned dataset
CleanData = JointData_knn_Imputer.copy()
CleanData.head(n=1000)

Diabetes_binary HighChol   BMI Smoker Stroke HeartDiseaseorAttack  \
0                  0      0.0  25.0    1.0    0.0                  0.0   
1                  0      1.0  19.0    0.0    0.0                  0.0   
2                  0      0.0  23.0    0.0    1.0                  0.0   
3                  1      1.0  35.0    0.0    0.0                  0.0   
4                  0      0.0  31.0    0.0    0.0                  0.0   
..               ...      ...   ...    ...    ...                  ...   
995                0      0.0  43.4    0.0    0.0                  0.0   
996                1      0.0  40.0    0.0    0.0                  0.0   
997                0      0.0  28.4    0.0    0.0                  0.0   
998                0      0.0  26.3    0.0    0.0                  0.0   
999                1      1.0  28.3    1.0    0.0                  0.0   

    PhysActivity HvyAlcoholConsump  MentHlth  PhysHlth DiffWalk AgeLevel  \
0            1.0               0.0         0        15      0.0     10.0   
1            0.0               0.0         0        14      0.0      9.0   
2            0.0               0.0         0        30      1.0     11.0   
3            0.0               0.0         0         0      1.0     12.0   
4            1.0               0.0         2        10      0.0      1.0   
..           ...               ...       ...       ...      ...      ...   
995          1.0               0.0        10         5      0.0      3.0   
996          1.0               0.0         7         4      0.0      1.0   
997          1.0               0.0         7         3      0.0      1.0   
998          1.0               0.0         6         1      0.0      1.0   
999          1.0               0.0        12         8      0.0      5.0   

     Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin  \
0              9      143             71             27      210   
1              5      134             70             25      122   
2              7      141             73             29      165   
3              6      152             82             37      214   
4              2      111             69             25      166   
..           ...      ...            ...            ...      ...   
995            0       93            100             39       72   
996            1      181             78             42      293   
997            2       88             58             26       16   
998            3      116             74             15      105   
999           11      136             84             35      130   

     DiabetesPedigreeFunction  
0                      0.5975  
1                      0.5288  
2                      0.5717  
3                      0.7138  
4                      0.6118  
..                        ...  
995                    1.0210  
996                    1.2580  
997                    0.7660  
998                    0.1070  
999                    0.2600  

[1000 rows x 18 columns]